src/Application/Gallery/Gallery.tsx

```diff
@@ -55,12 +55,19 @@ class Gallery extends React.Component<Props, State> {
   };
 
   componentDidMount() {
-    const { children, imagesDisplayed = defaultImagesDisplayed } = this.props;
+    const {
+      children,
+      imagesDisplayed = defaultImagesDisplayed,
+      isModalVisible = false,
```

Comment:
@ritikamotwani do you think this works?
``` 
constructor(props: Props) {
    super(props);
    this.sliderRef = React.createRef();
this.state = {
    visible: props.initialVisibility,
    currentIndex: 1,
    imageLeft: 0,
  };
  }
```

Reply:
No, it works fine in case when the modal is not visible initially. But for the other case, the 2nd image opens, and the arrow keys do not work when the modal opens on the initial render. When I do `currentIndex: 0`, then the first image opens correctly on the initial render, but the arrow keys do not work.
They start working when I close the modal, and reopen it by clicking on an image.

Reply:
The arrow keys are not working because the focus isn't set to the slider somehow, and the key down event isn't getting detected. When I click somewhere on the image, the slider window is focused and the onKeyDown event is handled again

Reply:
Had to remove the comparison of prevState visibility of false from `componentDidUpdate`. Updated it now.